packages/admin-app/src/components/ElectionSetupWizard/ElectionSetupWizard.tsx

```diff
@@ -64,26 +71,18 @@ export const ElectionSetupWizard: React.FC = () => {
             <WizardStep active={step === ElectionSetupStep.JointKeyRetrieved}>
                 <JointKeyUploadStep onNext={handleNext} onChanged={handleChanged} />
             </WizardStep>
-            <WizardStep active={step === ElectionSetupStep.ManifestMenu}>
-                <ManifestMenuStep
-                    onUploadManifest={() => setStep(ElectionSetupStep.ManifestUpload)}
-                />
-            </WizardStep>
             <WizardStep active={step === ElectionSetupStep.ManifestUpload}>
-                <ManifestUploadStep
-                    onNext={() => setStep(ElectionSetupStep.ManifestPreview)}
-                    uploadManifest={async () => true}
-                />
+                <ManifestUploadStep onNext={handleNext} onUploadManifest={handleUploadManifest} />
             </WizardStep>
             <WizardStep active={step === ElectionSetupStep.ManifestPreview}>
                 <ManifestPreviewStep
-                    onNext={handleNext}
-                    backToMenu={() => setStep(ElectionSetupStep.ManifestMenu)}
-                    preview={service.getManifestPreview()}
+                    onNext={handleSubmit}
+                    backToMenu={() => setStep(ElectionSetupStep.ManifestUpload)}
+                    preview={service.getManifestPreview(manifest, request)}
```

Comment:
One thing I dislike about this is that by putting it here getManifestPreview appears to get called 8 times throughout the course of a user going through the wizard when it only needs to get called if `step === ElectionSetupStep.ManifestPreview`.  Not sure if I'm doing React wrong, I would solve this in Angular with an `*ngIf`.

Reply:
For the preview step control could you use something like this:

` {step === ElectionSetupStep.ManifestPreview && <ManifestPreviewStep ....  />}`

Reply:
It shouldn't be getting called, but it may be getting referenced on refresh. Those are not the same thing. The WizardStep should already be doing the code @SteveMaier-IRT wrote.

Reply:
That fixed it, thanks @SteveMaier-IRT!